### PR TITLE
[#11260] Instructor viewing students: incorrect message flashed before populating the page

### LINKED ISSUE #11260: Instructor viewing students: incorrect message flashed before populating the page
Steps:
1. Login as an instructor
2. Click this option
   image

Problem: The next page shows a 'no students in this course' warning momentarily before populating the page, even if the course actually has students. 

Note: Might not be visible when using the dev server as the page will get populated quickly.

### PR DESCRIPTION
**Pull request by lyeyixian**
Fixes #11260 

**Outline of Solution**

Added a boolean `isStudentListEmpty` and check the variable in the `*ngIf` attribute instead of `students.length === 0`

Changed the if check `!sections.length` into `!Object.keys(sections).length` as sections doesn't have the length attribute and sections is undefined when the student list is empty which causes the buggy UI.

**Before**

https://user-images.githubusercontent.com/58477697/125494612-c146dc56-c948-4897-b1f5-576012140804.mp4

**Updated**

https://user-images.githubusercontent.com/58477697/125494626-316267f2-1486-4afa-a9d5-f4245e0af51a.mp4

### CODE REVIEW

**Review by daongochieu2810 — APPROVED** [review]
Changes look good to me!

File: src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts [hunk 1]
```diff
@@ -143,13 +145,18 @@ export class InstructorCourseDetailsPageComponent implements OnInit {
         this.loadPrivilege(courseid, key, data);
       });
 
-      if (!sections.length) {
-        this.isStudentsLoading = false;
+      if (!Object.keys(sections).length) {
```
Comment (ccyccyccy) [inline src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts:148]: I just tried this PR out. Looks like this single line fix alone can solve the issue. Are the other changes neccessary?

Comment (lyeyixian) [reply src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts:148]: Hmmm, yup seems like it. Initially, I thought that the check in the `*ngIf` was the issue. When I change from using `students.length === 0` to a boolean variable, it works also.

I will update it.

**Comment by rrtheonlyone** [comment]
@lyeyixian remember to fix the CI here before requesting a review from us.

**Comment by lyeyixian** [comment]
Oh, sorry. I didn't look closely and thought it was just the warning when running the test locally. I fixed it already.

File: src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts (hunk 1 above)
Comment (ccyccyccy) [reply src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts:148]: @lyeyixian Have you removed the unnecessary changes?

Comment (lyeyixian) [reply src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts:148]: Yes, I already did. You can see it in this commit https://github.com/TEAMMATES/teammates/pull/11271/commits/00d5f0dac86b6649bde712371abcab1d92fc5248

**Review by rrtheonlyone — APPROVED** [review]
Good fix @lyeyixian! Your solution works

I made some small tweaks to also prevent multiple toggles of the loading flag.

### Explanation of my additional change

Let's look at at these two functions that are called upon page load:
https://github.com/TEAMMATES/teammates/blob/812d0e66e1220b6f7a862cae52d7d6b0ced3465b/src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts#L119-L179

Let's focus on the variable `isStudentsLoading`. This is the variable that dictates what is displays either "no students" or the table when toggled. Sequence of events:
1. Loads students via API
2. Group students into section and call get privileges API to assign instructor privilege to student

We toggle the variable to false in step (2). However, we should only toggle it once all the section privileges have been handled and not immediately. This is not a big issue and is unlikely to cause big problems.

**Comment by lyeyixian** [comment]
Ah okok, I see. Thanks for letting me know.

**Pull request merged**